src/python/pants/backend/codegen/protobuf/java/protobuf_gen.py

```diff
@@ -82,12 +83,9 @@ def __init__(self, *args, **kwargs):
     self.plugins = self.get_options().protoc_plugins or []
     self._extra_paths = self.get_options().extra_path or []
 
-  @memoized_property
+  @property
   def protobuf_binary(self):
-    binary_util = BinaryUtil.Factory.create()
-    return binary_util.select_binary(self.get_options().supportdir,
-                                     self.get_options().version,
-                                     'protoc')
+    return Protoc.global_instance().select(context=self.context)
```

Comment:
Is there a way we can better guarantee that the subsystem dependency is actually declared? (e.g. adding a `subsystem_dependencies` instance method which returns the instantiated versions of the classes returned by `subsystem_dependencies`, maybe as a `dict<class,instance>`)

I can't think of a nice way of doing this, and I suspect no way of trying would be more likely to offer the guarantee because there's nothing stopping you from poking at the global instance anyway, but if you have any ideas, I'd be interested...

Reply:
That's a general gotcha with subsystems, but not one I want to tackle here.

Reply:
It will be a thing that we can solve by supporting Subsystems natively in v2... will be one of the first things we need to solve.